### PR TITLE
usm: watcher: Add periodic scan

### PR DESCRIPTION
**Pull request by vitkyrka**
### What does this PR do?

There are cases where the eBPF shared library watcher can miss hooking shared libraries in running processes.  For example:

 (1) When a process such as nginx daemonizes itself by forking.
 Depending on timing, the parent process may exit before the hooking is
 complete leading to the hooking of the library failing, and the child
 process will never be noticied by the watcher since it never opens the
 shared library (since it's inherited from the parent).

 (2) When the path length is longer than what the eBPF program supports.

To ensure that we don't miss these kind of cases, introduce a periodic scan to try to attach to running processes.  We maintain a map to ensure that we don't scan processes over and over again to limit CPU usage.  We scan every processes twice in order to avoid races with startup where we may scan a process' maps file before it has opened its shared libraries.

### Motivation

https://datadoghq.atlassian.net/wiki/spaces/UT/pages/4105764896

### Describe how you validated your changes

Automated test added.

Also verified manually that with the fix `nginx` is monitored if launched like this

```
$ docker run -v /proc:/proc -it --entrypoint /bin/bash nginx
root@1d70122a9e6c:/# nginx
``` 

By checking the debug endpoint `sudo curl --unix /opt/datadog-agent/run/sysprobe.sock http://unix/network_tracer/debug/usm/traced_programs  | jq | gron`

### Additional Notes

[Profile](https://ddstaging.datadoghq.com/profiling/explorer?query=service%3Asystem-probe%20kube_cluster_name%3Avenomoth%20version%3A7.62.0-devel_git.526.264e047&agg_m=count&agg_m_source=base&agg_t=count&my_code=disabled&op_filter=focus_on%28function%3A%22%28%2AWatcher%29.sync%22%29&profile_type=cpu-time&viz=flame_graph&start=1735030140000&end=1735033800000&paused=true) on deployment in staging. The sync function takes 0.18% of CPU time in this profile.

### CODE REVIEW

File: pkg/network/usm/sharedlibraries/watcher.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
-const (
+var (
```
Comment (guyarb) [inline pkg/network/usm/sharedlibraries/watcher.go:29]: why did you change it?

Comment (vitkyrka) [reply pkg/network/usm/sharedlibraries/watcher.go:29]: The test changes this to a much smaller value in order to avoid waiting for a long time.

Comment (guyarb) [reply pkg/network/usm/sharedlibraries/watcher.go:29]: Can you leave a comment in the code then?

File: pkg/network/usm/sharedlibraries/watcher.go [hunk 2]
```diff
@@ -291,28 +290,57 @@ func (w *Watcher) Start() {
 	utils.AddAttacher(consts.USMModuleName, "native", w)
 }
 
-// findDeletedProcesses returns the terminated PIDs from the given map.
-func findDeletedProcesses[V any](pids map[uint32]V) map[uint32]struct{} {
-	existingPids := make(map[uint32]struct{}, len(pids))
+func (w *Watcher) sync() {
```
Comment (guyarb) [inline pkg/network/usm/sharedlibraries/watcher.go:293]: Can you add a documentation for this function?

File: pkg/network/usm/sharedlibraries/watcher.go [hunk 3]
```diff
@@ -291,28 +290,57 @@ func (w *Watcher) Start() {
...
+		}
+
+		scanned := w.scannedPIDs[pid]
+
+		// Try to scan twice. This is because we may happen to scan the process
+		// just after it has been exec'd and before it has opened its shared
+		// libraries. Scanning twice with the sync interval reduce this risk of
+		// missing shared libraries due to this.
+		if scanned == 0 || scanned == 1 {
+			w.scannedPIDs[pid] = scanned + 1
```
Comment (guyarb) [inline pkg/network/usm/sharedlibraries/watcher.go:325]:
```suggestion
			w.scannedPIDs[pid]++
```

File: pkg/network/usm/sharedlibraries/watcher.go [hunk 4]
```diff
@@ -291,28 +290,57 @@ func (w *Watcher) Start() {
...
+			return nil
+		}
+
+		scanned := w.scannedPIDs[pid]
+
+		// Try to scan twice. This is because we may happen to scan the process
+		// just after it has been exec'd and before it has opened its shared
+		// libraries. Scanning twice with the sync interval reduce this risk of
+		// missing shared libraries due to this.
+		if scanned == 0 || scanned == 1 {
```
Comment (guyarb) [inline pkg/network/usm/sharedlibraries/watcher.go:324]:
```suggestion
		if scanned < 2 {
```

**Review by guyarb — APPROVED** [review]
LGTM, the only comments I "insist" about are the documentation, the rest are just suggestions
